modules/functional_expansion_tools/include/userobject/FXIntegralBaseUserObject.h

```diff
@@ -0,0 +1,246 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef FXINTEGRALBASEUSEROBJECT_H
+#define FXINTEGRALBASEUSEROBJECT_H
+
+// MOOSE includes
+#include "AuxiliarySystem.h"
+#include "MooseError.h"
+#include "MooseMesh.h"
+#include "MooseVariable.h"
+#include "UserObject.h"
+
+// libMesh includes
+#include "libmesh/quadrature.h"
+
+// Module includes
+#include "FunctionSeries.h"
+#include "MutableCoefficientsInterface.h"
+
+/**
+ * Class declaration for parameters - we cannot use templated types in validParams<>()
```

Comment:
I tried something similar. At least it didn't work since I'm providing template parameter, not an actual type. Maybe I did it wrong?

`FXIntegralBaseUserObject` is a host class that provides a policy-based design. This base object generically allows a functional expansion to be derived from any `UserObject` that contains `computeQpIntegral()` as the policy class.

The same base `InputParameters` are required, no matter what policy is provided (although each policy may need to add some more like `FXBoundaryFluxUserObject`). I'd really hate to copy-paste the parameters for each policy. This is the reason that I configured an empty parameter class. I'm very open to a better solution.